mern-ecommerce/controllers/auth.js

```diff
@@ -80,3 +80,81 @@ exports.isAdmin = (req, res, next) => {
   }
   next();
 };
+
+exports.checkUser = (req, res, next) => {
+  let user = req.profile && req.auth && req.profile._id == req.auth._id;
+  if (!user) {
+    return res.status(403).jssosn({
+      error: 'Access denied',
+    });
+  }
+  next();
+};
+
+exports.forgotPassword = (req, res) => {
+  if (!req.body) return res.status(400).json({ message: 'No request body' });
+  if (!req.body.email)
+    return res.status(400).json({ message: 'No Email in request body' });
+
+  console.log('forgot password finding user with that email');
+  const { email } = req.body;
+  console.log('signin req.body', email);
+
+  User.findOne({ email }, (err, user) => {
+    if (err || !user)
```

Comment:
This condition is not checking if `user` is undefined, it only checks if `user` is falsy. It should be `if (err || typeof user === 'undefined' || !user)`.